simd/keccakf1600/example_test.go

```diff
@@ -1,79 +1,66 @@
 package keccakf1600_test
 
 import (
-	"github.com/cloudflare/circl/simd/keccakf1600"
-
 	"encoding/binary"
 	"fmt"
 
-	// Vendored copy of golang.org/x/crypto/sha3.
-	sha3 "github.com/cloudflare/circl/internal/shake"
+	"github.com/cloudflare/circl/simd/keccakf1600"
 )
 
-func Example() {
+func Example_keccakf1600() {
 	// As an example, computes the (first 32 bytes of a) SHAKE-256 stream of
 	// four short strings at the same time.
-	// See sign/dilithium/mode3/internal/sample.go for a real world example.
-
-	msgs := [][]byte{
+	msgs := [4][]byte{
 		[]byte("These are some short"),
 		[]byte("strings of the same "),
 		[]byte("length that fit in a"),
 		[]byte("single block.       "),
 	}
-	var hashes [4][32]byte
 
-	if !keccakf1600.AvailableX4 {
-		// Compute hashes separately using golang.org/x/crypto/sha3 instead
-		// when a fast four-way implementation is not available.  A generic
-		// keccakf1600 implementation is quite a bit slower than using
-		// the non-interleaved hashes because of the need to interleave and
-		// deinterleave the state.
-		for i := 0; i < 4; i++ {
-			h := sha3.NewShake256()
-			_, _ = h.Write(msgs[i])
-			_, _ = h.Read(hashes[i][:])
-		}
-	} else {
-		// f1600 acts on 1600 bits arranged as 25 uint64s.  Our fourway f1600
-		// acts on four interleaved states; that is a [100]uint64.  (A separate
-		// type is used to ensure that the encapsulated [100]uint64 is aligned
-		// properly to be used efficiently with vector instructions.)
-		var perm keccakf1600.StateX4
-		state := perm.Initialize()
+	// The user could branch to a fast non-SIMD implementation if this function
+	// returns false.
+	_ = keccakf1600.IsSIMDEnabled()
```

Comment:
Not showing the branch in the example good gives a bad example: they should branch on it!